src/test_l1_dsFPD.c

```diff
@@ -667,68 +730,99 @@ void test_l1_dsFPD_positive_dsSetFPBrightness (void)
     result = dsFPInit();
-    UT_ASSERT_EQUAL(result, dsERR_NONE);
+    UT_ASSERT_EQUAL_FATAL(result, dsERR_NONE);
 
     // Variation 03: Set all valid indicators to dsFPD_STATE_ON using dsSetFPState()
     // Variation 04: Pass an invalid eBrightness parameter to dsSetFPBrightness()
-    for (int i = 0; i < sizeof(kIndicators) / sizeof(kIndicators[0]); ++i)
+    for(int i = 1; i <= count; i++)
     {
-        UT_LOG("\n In %s , Indicator: [%d]\n", __FUNCTION__, kIndicators[i].id);
-        result = dsSetFPState(kIndicators[i].id, dsFPD_STATE_ON);
+	snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/Indicator_Type", i);
+        eIndicator = UT_KVP_PROFILE_GET_UINT32(buffer);
+        UT_LOG("\n In %s , Indicator: [%d]\n", __FUNCTION__, eIndicator);
+        result = dsSetFPState(eIndicator, dsFPD_STATE_ON);
         UT_ASSERT_EQUAL(result, dsERR_NONE);
-        result = dsSetFPBrightness(kIndicators[i].id, 200);
+        result = dsSetFPBrightness(eIndicator, 200);
         UT_ASSERT_EQUAL(result, dsERR_INVALID_PARAM);
     }
 
-    // Variation 05: Pass an invalid eIndicator parameter to dsSetFPBrightness()
-    result = dsSetFPBrightness(dsFPD_INDICATOR_MAX, 50);
-    UT_ASSERT_EQUAL(result, dsERR_INVALID_PARAM);
-    
-    // Variation 06: Set all valid indicators to dsFPD_STATE_OFF using dsSetFPState()
-    for (int i = 0; i < sizeof(kIndicators) / sizeof(kIndicators[0]); ++i)
+    // Variation 05: Set all valid indicators to dsFPD_STATE_OFF using dsSetFPState()
+    for(int i = 1; i <= count; i++)
     {
-        result = dsSetFPState(kIndicators[i].id, dsFPD_STATE_OFF);
+        snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/Indicator_Type", i);
+        eIndicator = UT_KVP_PROFILE_GET_UINT32(buffer);
+        result = dsSetFPState(eIndicator, dsFPD_STATE_OFF);
         UT_ASSERT_EQUAL(result, dsERR_NONE);
     }
 
-    // Variation 07: Call dsSetFPBrightness() and loop through all valid indicators from kIndicators
-    for (int i = 0; i < sizeof(kIndicators) / sizeof(kIndicators[0]); ++i)
+    // Variation 06: Call dsSetFPBrightness() and loop through all valid indicators from yaml
+    for(int i = 1; i <= count; i++)
     {
-        result = dsSetFPBrightness(kIndicators[i].id, 50);
+        snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/Indicator_Type", i);
+        eIndicator = UT_KVP_PROFILE_GET_UINT32(buffer);
+	snprintf(minbuffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/MIN_BRIGHTNESS",eIndicator);
+        minBrightness =  UT_KVP_PROFILE_GET_UINT32(minbuffer);
+        snprintf(maxbuffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/MAX_BRIGHTNESS",eIndicator);
+        maxBrightness = UT_KVP_PROFILE_GET_UINT32(maxbuffer);
+        int avgBrightness = (maxBrightness + minBrightness) / 2;
+        result = dsSetFPState(eIndicator, dsFPD_STATE_ON);
+        UT_ASSERT_EQUAL(result, dsERR_NONE);
+        result = dsSetFPBrightness(eIndicator, avgBrightness);
```

Comment:
Why are we validating with NOT_SUPORTED here where we have kept a valid value?  Can you please look into it?  Also the value LED is not put to OFF state after use (setting the FPBrightness).

Reply:
Done